src/app/component/quiz/quiz.component.ts

```diff
@@ -156,75 +144,67 @@ export class Quizcomponent implements OnInit, OnDestroy {
     this.router.navigate(['/result']);
   }
 
-  skip(questionindex: number) {
+  skipQuestion(questionIndex: number) {
     let label = this.dialogData.skipModel.label;
     let yesButtonLable = this.dialogData.skipModel.yesButtonLable;
     let NoButtonLable = this.dialogData.skipModel.NoButtonLable;
     this.dialogService
       .openDialog(label, yesButtonLable, NoButtonLable)
       .then((value) => {
         if (value) {
-          this.questionindex = questionindex;
-          this.next(this.questionindex);
+          this.questionIndex = questionIndex;
+          this.nextQuestion(this.questionIndex);
         }
       });
   }
 
   disabledValuesAndForm() {
     this.FormArray.controls
-      .at(this.questionindex)
+      .at(this.questionIndex)
       ?.get('radioValue')
       ?.disable();
-    this.FormArray.controls.at(this.questionindex)?.get('timer')?.disable();
-    this.FormArray.controls.at(this.questionindex)?.markAsDirty();
+    this.FormArray.controls.at(this.questionIndex)?.get('timer')?.disable();
+    this.FormArray.controls.at(this.questionIndex)?.markAsDirty();
   }
 
   startCounter() {
     this.interval$ = interval(1000)
       .pipe(takeUntil(this.destroyed$))
       .subscribe((val) => {
-        const counterValue = this.FormArray.controls.at(this.questionindex)
+        const counterValue = this.FormArray.controls.at(this.questionIndex)
           ?.value.timer;
         if (
           Number(counterValue) != 0 &&
-          !this.FormArray.controls.at(this.questionindex)?.get('timer')
+          !this.FormArray.controls.at(this.questionIndex)?.get('timer')
             ?.disabled
         ) {
           this.FormArray.controls
-            .at(this.questionindex)
+            .at(this.questionIndex)
             ?.patchValue({ timer: counterValue - 1 });
         } else if (
           Number(counterValue) === 0 &&
-          !this.FormArray.controls.at(this.questionindex)?.get('timer')
+          !this.FormArray.controls.at(this.questionIndex)?.get('timer')
             ?.disabled
         ) {
-          this.next(this.questionindex);
+          this.nextQuestion(this.questionIndex);
         }
       });
   }
 
-  answer(questionindex: number, correctOptions: string) {
-    if (!this.FormArray.at(questionindex).get('timer')?.disabled) {
-      if (questionindex === this.question.length) {
-        this.isQuizCompleted = true;
-      }
-      let selectedAnswer =
-        this.question[questionindex].answer?.[0] == correctOptions;
-      if (selectedAnswer && correctOptions) {
-        this.points = this.points += 1;
+  answer(questionIndex: number, correctOptions: string) {
+    if (!this.FormArray.at(questionIndex).get('timer')?.disabled) {
+      if (this.question[questionIndex].answer.id && correctOptions) {
```

Comment:
why condition is like this.question[questionIndex].answer.id && correctOption 
 It should be suppose to this.question[questionindex].answer.id == correctOptions 
 And please change correctOptions to correctOption

Reply:
Better if we change name to selectedOption

Reply:
yes i have changed the name and also wrote the proper condition